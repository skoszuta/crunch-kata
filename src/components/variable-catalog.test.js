import fs from 'fs';
import React from 'react';
import { mount, render } from 'enzyme';
import 'jest-styled-components';

import VariableCatalog from './variable-catalog';

const { index } = JSON.parse(fs.readFileSync('json/variables.json', 'utf8'));
const { graph: order } = JSON.parse(fs.readFileSync('json/order.json', 'utf8'));

describe('VariableCatalog', () => {
  const variables = { index, order };

  describe('snapshot', () => {
    it('should render correctly', () => {
      const wrapper = render(
        <VariableCatalog variables={variables} />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = mount(
        <VariableCatalog variables={variables} />
      );
    });

    it('should render all sub-trees initially collapsed', () => {
      const subTrees = wrapper.find('SubTree');

      subTrees.forEach(subTree => expect(subTree).toHaveStyleRule('display', 'none'));
    });

    // Below tests disabled due to enzyme wrapper not being updated after calling simulate()
    // See: https://github.com/airbnb/enzyme/issues/1400
    //
    // Below tests reference the .test-category-wrapper class which should be removed as soon as
    // https://github.com/airbnb/enzyme/issues/410 gets resolved
    //
    // it('should uncollapse a category sub-tree on click', () => {
    //   const categoryWrapper = wrapper.find('.test-category-wrapper').at(0);
    //   const button = categoryWrapper.find('Category').at(0);
    //   const subtree = categoryWrapper.find('SubTree').at(0);
    //
    //   button.simulate('click');
    //
    //   expect(subtree).not.toHaveStyleRule('display', 'none');
    // });
    //
    // it('should collapse a category sub-tree on click', () => {
    //   const categoryWrapper = wrapper.find('.test-category-wrapper').at(0);
    //   const button = categoryWrapper.find('Category').at(0);
    //   const subtree = categoryWrapper.find('SubTree').at(0);
    //
    //   button.simulate('click');
    //   button.simulate('click');
    //
    //   expect(subtree).toHaveStyleRule('display', 'none');
    // });
  });
});
