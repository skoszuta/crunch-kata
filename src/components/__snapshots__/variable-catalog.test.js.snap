// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariableCatalog snapshot should render correctly 1`] = `
<styled.ul>
  <li
    className="test-category-wrapper"
    key="Awareness Metrics"
  >
    <Category
      collapsed={true}
      depth={0}
      onClick={[Function]}
    >
      Awareness Metrics
    </Category>
    <SubTree
      collapsed={true}
    >
      <li
        className="test-category-wrapper"
        key="Taxis"
      >
        <Category
          collapsed={true}
          depth={1}
          onClick={[Function]}
        >
          Taxis
        </Category>
        <SubTree
          collapsed={true}
        >
          <li
            key="11b0b9"
          >
            <styled.button
              depth={2}
            >
              Unaided Awareness (Taxis)
            </styled.button>
          </li>
          <li
            key="7a89e0"
          >
            <styled.button
              depth={2}
            >
              Aided Awareness (Taxis)
            </styled.button>
          </li>
        </SubTree>
      </li>
      <li
        className="test-category-wrapper"
        key="Coffee"
      >
        <Category
          collapsed={true}
          depth={1}
          onClick={[Function]}
        >
          Coffee
        </Category>
        <SubTree
          collapsed={true}
        >
          <li
            key="786c0f"
          >
            <styled.button
              depth={2}
            >
              Unaided Awareness (Coffee Roasters)
            </styled.button>
          </li>
          <li
            key="2d27ab"
          >
            <styled.button
              depth={2}
            >
              Aided Awareness (Coffee Roasters)
            </styled.button>
          </li>
        </SubTree>
      </li>
    </SubTree>
  </li>
  <li
    className="test-category-wrapper"
    key="Purchase Consideration and Behavior"
  >
    <Category
      collapsed={true}
      depth={0}
      onClick={[Function]}
    >
      Purchase Consideration and Behavior
    </Category>
    <SubTree
      collapsed={true}
    >
      <li
        className="test-category-wrapper"
        key="Taxis"
      >
        <Category
          collapsed={true}
          depth={1}
          onClick={[Function]}
        >
          Taxis
        </Category>
        <SubTree
          collapsed={true}
        >
          <li
            key="ede6a8"
          >
            <styled.button
              depth={2}
            >
              Consideration
            </styled.button>
          </li>
          <li
            key="62c00f"
          >
            <styled.button
              depth={2}
            >
              Likely Purchase
            </styled.button>
          </li>
          <li
            key="0f6ce0"
          >
            <styled.button
              depth={2}
            >
              Usage
            </styled.button>
          </li>
          <li
            key="f2a681"
          >
            <styled.button
              depth={2}
            >
              Specific Service Usage
            </styled.button>
          </li>
        </SubTree>
      </li>
      <li
        className="test-category-wrapper"
        key="Coffee"
      >
        <Category
          collapsed={true}
          depth={1}
          onClick={[Function]}
        >
          Coffee
        </Category>
        <SubTree
          collapsed={true}
        >
          <li
            key="8d7127"
          >
            <styled.button
              depth={2}
            >
              Blue Bottle Coffee Usage
            </styled.button>
          </li>
          <li
            key="ee3e40"
          >
            <styled.button
              depth={2}
            >
              Stumptown Coffee Roasters Usage
            </styled.button>
          </li>
          <li
            key="8f7db6"
          >
            <styled.button
              depth={2}
            >
              Ritual Roasters Usage
            </styled.button>
          </li>
          <li
            key="3a5f89"
          >
            <styled.button
              depth={2}
            >
              FOURBARREL Usage
            </styled.button>
          </li>
        </SubTree>
      </li>
    </SubTree>
  </li>
  <li
    className="test-category-wrapper"
    key="Demographics and Technographics"
  >
    <Category
      collapsed={true}
      depth={0}
      onClick={[Function]}
    >
      Demographics and Technographics
    </Category>
    <SubTree
      collapsed={true}
    >
      <li
        key="c83da2"
      >
        <styled.button
          depth={1}
        >
          Gender
        </styled.button>
      </li>
      <li
        key="e08fbc"
      >
        <styled.button
          depth={1}
        >
          Education
        </styled.button>
      </li>
      <li
        key="1b44a4"
      >
        <styled.button
          depth={1}
        >
          Marital Status
        </styled.button>
      </li>
      <li
        key="5b6a73"
      >
        <styled.button
          depth={1}
        >
          Children Under 18
        </styled.button>
      </li>
      <li
        key="91a33f"
      >
        <styled.button
          depth={1}
        >
          Family Income
        </styled.button>
      </li>
      <li
        key="9f4b60"
      >
        <styled.button
          depth={1}
        >
          (unnamed variable)
        </styled.button>
      </li>
    </SubTree>
  </li>
  <li
    className="test-category-wrapper"
    key="Dimensions"
  >
    <Category
      collapsed={true}
      depth={0}
      onClick={[Function]}
    >
      Dimensions
    </Category>
    <SubTree
      collapsed={true}
    >
      <li
        key="d88613"
      >
        <styled.button
          depth={1}
        >
          City
        </styled.button>
      </li>
      <li
        key="0323cf"
      >
        <styled.button
          depth={1}
        >
          Wave
        </styled.button>
      </li>
    </SubTree>
  </li>
  <li
    key="d0fe8b"
  >
    <styled.button
      depth={0}
    >
      weight
    </styled.button>
  </li>
  <li
    key="0894c5"
  >
    <styled.button
      depth={0}
    >
      Weight
    </styled.button>
  </li>
</styled.ul>
`;
